src/sersicmix.cc

```diff
@@ -0,0 +1,2091 @@
+#include "sersicmix.h"
+
+#include <array>
+#include <stdexcept>
+#include <vector>
+
+#include "util.h"
+
+namespace gauss2d
+{
+namespace fit
+{
+
+std::string IntegralSize::str() const {
+    return "IntegralSize(integral=" + std::to_string(integral) + ", sigma=" + std::to_string(sigma) + ")";
+}
+IntegralSize::IntegralSize(double integral_, double sigma_) : integral(integral_), sigma(sigma_)  {}
+
+std::string SersicMixValues::str() const {
+    return "SersicMixValues(sersicindex=" + std::to_string(sersicindex) + ", values=" + str_iter_ref(values) + ")";
+}
+SersicMixValues::SersicMixValues(double sersicindex_, std::vector<IntegralSize> values_)
+    : sersicindex(sersicindex_), values(values_)  {}
+
+const std::vector<SersicMixValues> & get_sersic_mix_knots(unsigned short order) {
+    if(order == 4) return get_sersic_mix_knots_order4();
+    else if (order == 8) return get_sersic_mix_knots_order8();
+    throw std::invalid_argument("Unsupported Sersic mixture knot order=" + std::to_string(order));
+}
+
+const std::vector<SersicMixValues> & get_sersic_mix_knots_order4() {
+    static const std::vector<SersicMixValues> values = get_sersic_mix_knots_copy(4);
+    return values;
+}
+
+const std::vector<SersicMixValues> & get_sersic_mix_knots_order8() {
+    static const std::vector<SersicMixValues> values = get_sersic_mix_knots_copy(8);
+    return values;
+}
+
+/*
+Converting from old multiprofit:
+
+nl = '\n'
+for key, value in values.items():
+    norm_strs = [f"{x:.15e}" for x in value[0]]
+    value[0][-1] = 1.0 - sum(float(x) for x in norm_strs[:-1])
+    norm_strs[-1] = f"{value[0][-1]:.15e}"
+    value_sum = sum(float(x) for x in norm_strs)
+    if value_sum != 1:
+        raise ValueError(f'key={key} value_sum-1={value_sum-1} != 0')
+    print(f'{{{key}, {{\n    {f",{nl}    ".join(f"{{{x}, {y*gauss2d.M_HWHM_SIGMA}}}" for x, y in zip(norm_strs, value[1]))}{nl}}}}},')
+*/
```

Comment:
I'm guessing that this is the script that you used to generate the text in the rest of the file, right? If so, it might be better to save this script somewhere else rather than leave it as a comment here.